src/backend/distributed/utils/node_metadata.c

```diff
@@ -102,6 +107,11 @@ master_remove_node(PG_FUNCTION_ARGS)
 							   "shard placements")));
 	}
 
+	workerNode = FindWorkerNode(nodeNameString, nodePort);
+
+	nodeDeleteCommand = NodeDeleteCommand(workerNode->nodeId);
+	SendCommandToWorkers(WORKERS_WITH_METADATA, nodeDeleteCommand);
```

Comment:
This would error out if the node itself has metadata and has already been taken down (e.g. we're scaling down the cluster). It might make sense to remove the pg_dist_node record before sending the command.